watertap/property_models/unit_specific/activated_sludge/tests/test_asm1_reaction.py

```diff
@@ -161,6 +167,121 @@ def check_units(self, model):
         assert_units_consistent(model)
 
 
+class TestASM1ReactionScaler(object):
```

Comment:
One thought that comes to mind--- I am hoping we can at least track some metric, even it doesn't adequately represent the shift in model states, to keep some record to quantify the performance of a scaler class (as well as the old calculate_scaling_factors method). 

Perhaps the easiest thing to do is to use the DiagnosticToolbox, check the condition number and assert that value for the new Scaler, the old method-based approach, and any new Scaler classes (or revisions to existing ones) that we bring in. This way, we can get some idea of how impactful scaler additions/revisions are as we progress.

Reply:
@adam-a-a I should be able to add this relatively easily. Can you take a look at the AD tests I added here for comparing the old and new scaling methods: https://github.com/watertap-org/watertap/blob/2e9aaafa0e020e8afa8a7beaf9e3e6ec37e59a00/watertap/unit_models/tests/test_anaerobic_digester.py#L854

That PR is still a WIP, but all I need to do is refine the scaling for the bio_P=False case in the BSM2_P flowsheet